File: backend/app/schemas/report.py
"""
Financial report schemas.
"""
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class DocumentTypeSummary(BaseModel):
    """Summary for a specific document type."""
    type: str
    count: int
    total: float


class PeriodReport(BaseModel):
    """Financial report for a time period."""
    period_start: datetime
    period_end: datetime
    total_invoiced: float
    total_quotes: float
    total_paid: float
    total_pending: float
    document_count: int
    by_type: List[DocumentTypeSummary]


class AnnualReport(BaseModel):
    """Annual report with monthly breakdown."""
    year: int
    total_invoiced: float
    total_paid: float
    total_pending: float
    months: List[PeriodReport]

File: frontend/src/types/report.ts
export interface DocumentTypeSummary {
  type: string;
  count: number;
  total: number;
}

export interface PeriodReport {
  period_start: string;
  period_end: string;
  total_invoiced: number;
  total_quotes: number;
  total_paid: number;
  total_pending: number;
  document_count: number;
  by_type: DocumentTypeSummary[];
}

export interface AnnualReport {
  year: number;
  total_invoiced: number;
  total_paid: number;
  total_pending: number;
  months: PeriodReport[];
}
